test/test_database.py

```diff
@@ -132,7 +131,8 @@ def test_create_collection(self):
         db = Database(self.client, "pymongo_test")
 
         db.test.insert_one({"hello": "world"})
-        self.assertRaises(CollectionInvalid, db.create_collection, "test")
+        if not client_context.is_topology_type(["sharded"]):
```

Comment:
`client_context.is_topology_type(["sharded"])` -> `client_context.is_mongos`

It's the same thing, the second is just less verbose.